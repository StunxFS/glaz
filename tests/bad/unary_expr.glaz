fn main() {
  let d = 10;
  _ = !d;
  let unsigned: u8 = 1;
  _ = -unsigned;
}
