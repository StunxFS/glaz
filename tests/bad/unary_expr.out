tests/bad/unary_expr.glaz:3:8: error: `!` can only be used with booleans
     | 
   3 |   _ = !d;
     |       ^^
tests/bad/unary_expr.glaz:5:8: error: cannot apply unary operator `-` to type `u8`
     | 
   5 |   _ = -unsigned;
     |       ^^^^^^^^^
     = note: unsigned values cannot be negated
glaz error: could not compile `unary_expr`, aborting due to 2 previous errors.
